source/userguide/auto_annotator.rst

```diff
@@ -0,0 +1,489 @@
+	}
+
+The rule above would annotate all text files named "train" or "test" in the "faq" and "salary" domains. Only sys_amount-of-money and sys_time entities would be annotated.
+Internally, the above rule is combined to a single pattern: "(faq|salary)/.*/(train.txt|test.txt)" and this pattern is matched against all file paths in the domain folder of your Mindmeld application. 
+
+.. warning::
+
+	The order of the annotation rules matters. Each rule overwrites the list of entities to annotate for a file if the two rules include the same file. It is good practice to start with more generic rules first and then have more specific rules.
+	Be sure to use the regex "or" (:attr:`|`) if applying rules at the same level of specificity. Otherwise, if written as separate rules, the latter will overwrite the former.
+
+.. warning::
+	By default, all files in all intents across all domains will be annotated with all supported entities. Before annotating consider including custom annotation rules in :attr:`config.py`. 
+
+``'unannotate_supported_entities_only'`` (:class:`boolean`): By default, when the unannotate command is used only entities that the Annotator can annotate will be eligible for removal. 
+
+``'unannotate'`` (:class:`list`): List of annotation rules in the same format as those used for annotation. These rules specify which entities should have their annotations removed. By default, :attr:`files` is None.
+
+``'spacy_model'`` (:class:`str`): :attr:`en_core_web_lg` is used by default for the best performance. Alternative options are :attr:`en_core_web_sm` and :attr:`en_core_web_md`. This parameter is optional and is specific to the use of the :class:`SpacyAnnotator`.
+If the selected English model is not in the current environment it will automatically be downloaded. Refer to Spacy's documentation to learn more about their `English models <https://spacy.io/models/en>`_.
+
+
+Creating a Custom Annotator
+---------------------------
+The :class:`SpacyAnnotator` is a subclass of the abstract base class :class:`Annotator`.
+The functionality for annotating and unannotating files is contained in :class:`Annotator` itself.
+A developer simply needs to implement two methods to create a custom annotator.
+
+
+Custom Annotator Boilerplate Code
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+This section includes boilerplate code to build a :class:`CustomAnnotator` class to which you can add to your own python file, let's call it :attr:`custom_annotator.py`
+There are two "TODO"s. To implement a :class:`CustomAnnotator` class a developer has to implement the :meth:`parse` and :meth:`valid_entity_check` methods.
+
+.. code-block:: python
+
+	class CustomAnnotator(Annotator):
+		""" Custom Annotator class used to generate annotations.
+		"""
+
+		def __init__(self, app_path, config=None):
+			super().__init__(app_path=app_path, config=config)
+			
+			# Add additional attributes if needed
+
+		def parse(self, sentence, entity_types=None):
```

Comment:
where do the entity types come from? probably we should make it clear that these entity types are the ones we want to annotate

Reply:
Updating the docs for this function:

```
        Args:
            sentence (str): Sentence to detect entities.
            entity_types (list): List of entity types to annotate. If None, all
                possible entity types will be annotated.
```